keps/sig-architecture/1544-kep-artifacts/README.md

```diff
@@ -0,0 +1,176 @@
+# KEP-1544: Enhance KEP artifacts for human and automatic consumption
+
+## Table of Contents
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Implementation Details/Notes/Constraints (optional)](#implementation-detailsnotesconstraints-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Examples](#examples)
+      - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+      - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+      - [Removing a deprecated flag](#removing-a-deprecated-flag)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Drawbacks (optional)](#drawbacks-optional)
+- [Alternatives (optional)](#alternatives-optional)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
```

Comment:
Might be out of scope for this KEP, but I feel like the release signoff checklist is under utilized, and just ends up being noise in the KEPs. I think part of the problem is that updating it requires sending a PR, and it might be better served as a checklist in the enhancement issue.

Reply:
out of scope for this, but I agree

Reply:
@tallclair @thockin -- This has to do w/ repo permissions (ref: https://github.com/kubernetes/enhancements/issues/590).
The person who opens the Enhancement issue may not necessarily be the person who carries it to the next release cycle, which means they'd have to have write access to the repo to update the Release Team Checklist.

Having the checklist diff in an issue also means you have to go to the issue and dig through the comment diffs. It's far easier for the @kubernetes/release-team to reconcile what was/wasn't done if they're in git history.

Definitely out of scope here, but the idea we were floating in SIG PM/Release was to have something of a receipts system.

Directory structure would look something like this:
```tree
releases
├── 1.16
│   ├── accepted
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   ├── exceptions
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   └── proposed
│       ├── 0001.yaml
│       ├── 0002.yaml
│       └── 0003.yaml
├── 1.17
│   ├── accepted
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   ├── exceptions
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   └── proposed
│       ├── 0001.yaml
│       ├── 0002.yaml
│       └── 0003.yaml
├── 1.18
│   ├── accepted
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   ├── exceptions
│   │   ├── 0001.yaml
│   │   ├── 0002.yaml
│   │   └── 0003.yaml
│   └── proposed
│       ├── 0001.yaml
│       ├── 0002.yaml
│       └── 0003.yaml
└── 1.19
    ├── accepted
    │   ├── 0001.yaml
    │   ├── 0002.yaml
    │   └── 0003.yaml
    ├── exceptions
    │   ├── 0001.yaml
    │   ├── 0002.yaml
    │   └── 0003.yaml
    └── proposed
        ├── 0001.yaml
        ├── 0002.yaml
        └── 0003.yaml
```

The release receipt would essentially be a SIG's commit to an enhancement for a cycle.
You'd PR a receipt to the `proposed/` directory of the release. Once Enhancements Freeze hits, some tools would scrape what's in the `proposed/` directory and move it to `accepted/`.

This would move the Enhancements team away from parsing Enhancement issues and over to approving PRs in a format that could tie to automation, instead of manually entered in a spreadsheet.

ref: https://github.com/kubernetes/sig-release/issues/539, https://github.com/kubernetes/sig-release/issues/432, https://github.com/kubernetes/sig-release/issues/486, https://github.com/kubernetes/sig-release/issues/493